src/globalstate.h

```diff
@@ -191,25 +222,45 @@ typedef struct GlobalState
   void **FreeList[MAX_GC_PREFIX_DESC+2];
 #endif
   /* Extra storage */
-} GlobalState;
+#if defined(HPCGAP)
+  void *Extra[TLS_NUM_EXTRA];
+#endif
+} GAPState;
 
-extern GlobalState *MainGlobalState;
+#if defined(HPCGAP)
 
-void InitMainGlobalState(void);
+#define TLS(x) (realTLS->x)
 
-void InitScannerState(GlobalState *);
-void InitStatState(GlobalState *);
-void InitExprState(GlobalState *);
-void InitCoderState(GlobalState *);
-void InitOpersState(GlobalState *);
+#else
 
-void DestroyScannerState(GlobalState *);
-void DestroyStatState(GlobalState *);
-void DestroyExprState(GlobalState *);
-void DestroyCoderState(GlobalState *);
-void DestroyOpersState(GlobalState *);
+#define TLS(x) (MainGAPState->x)
+extern GAPState *MainGAPState;
 
-void InitGlobalState(GlobalState *state);
-void DestroyGlobalState(GlobalState *state);
+#endif
+
+void InitMainGAPState(void);
+
+void InitScannerState(GAPState *);
+void InitStatState(GAPState *);
+void InitExprState(GAPState *);
+void InitCoderState(GAPState *);
+void InitOpersState(GAPState *);
+
+void DestroyScannerState(GAPState *);
+void DestroyStatState(GAPState *);
+void DestroyExprState(GAPState *);
+void DestroyCoderState(GAPState *);
+void DestroyOpersState(GAPState *);
+
+void InitGAPState(GAPState *state);
+void DestroyGAPState(GAPState *state);
+
+#if defined(HPCGAP)
+void InitAObjectsState(GAPState *state);
+void InitThreadAPIState(GAPState *state);
+
+void DestroyObjectsState(GAPState *state);
+void DestroyThreadAPIState(GAPState *state);
+#endif
```

Comment:
I guess the above decls should eventually be moved into their respective header files?